### PR TITLE
Prometheus: Fix instant query to run two times when exemplars enabled

### LINKED ISSUE #32485: Prometheus data somehow duplicated when Instant enabled
We recently updated Grafana from 7.4.3 to 7.5.0 and I've noticed new weird behavior with Prometheus queries, using a Table format and enabling the Instant button.
It looks like the data that is received from a single query is somehow duplicated.
I do expect that a single query will return 1 set of timeseries, and when setting the Table format + enabling Instant it should give me 1 resultset instead of the same resultset twice.

You can reproduce this on the Grafana Playground as well.

1. Go to: https://play.grafana.org/d/000000012/grafana-play-home?orgId=1
2. Add an empty panel (you can leave the visualization as Graph)
3. Select Prometheus as the DataSource
4. Use the following query: `prometheus_build_info{instance="demo.robustperception.io:9090"}`
5. Check the Graph and you'll see only 1 entry
6. Select the Table format underneath the query and watch the graph, you'll see you still have a single entry
7. Now enable the Instant button and you'll notice there's 2 times the same entries

For any tables I created in 7.4.3 it still works, refreshing the panel to get new datapoints won't duplicate the data.
However, if I select the input field of the query and then click outside the input field it immediately shows the same behavior as described above.

I just rolled back to 7.4.3 and this issue is no longer there.

I have not tested this against other Datasources because I don't have any to test it with :-)

### PR DESCRIPTION
**Pull request by zoltanbedi**
**What this PR does / why we need it**:

This PR fixes 2 problems.
1. In dashboard when instant query was selected it run 2 times
2. In dashboard changing the exemplars field with the eye icon didn't trigger run query

There is still an issue I found but couldn't fix it so I'm going to open an [issue](https://github.com/grafana/grafana/issues/32518) for that instead.

**Which issue(s) this PR fixes**:

Fixes #32485

### CODE REVIEW

File: public/app/plugins/datasource/prometheus/datasource.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ import { PrometheusVariableSupport } from './variables';
 import PrometheusMetricFindQuery from './metric_find_query';
 
 export const ANNOTATION_QUERY_STEP_DEFAULT = '60s';
+const EXEMPLARS_NOT_SUPPORTED = 'Exemplars are not supported in this version of prometheus.';
```
Comment (ifrost) [inline public/app/plugins/datasource/prometheus/datasource.ts:47]: I'm thinking of a case when Prometheus supports exemplars but the endpoint fails some other reason and this message could be somehow misleading. Would it make sense to make it more generic, e.g. "Exemplars for this data source are not available."?

**Review by ifrost — APPROVED** [review]
LGTM 🚀 . Tested on TNS. I just left a few minor comments about error messages.

File: public/app/plugins/datasource/prometheus/datasource.ts [hunk 2]
```diff
@@ -238,12 +239,17 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
...
+        if (target.exemplar && !target.instant) {
           const exemplarTarget = cloneDeep(target);
           exemplarTarget.requestId += '_exemplar';
           target.exemplar = false;
           queries.push(this.createQuery(exemplarTarget, options, start, end));
           activeTargets.push(exemplarTarget);
+          this.exemplarErrors.next();
+        }
+        if (target.exemplar && target.instant) {
+          this.exemplarErrors.next("Exemplars doesn't make sense with instant query.");
```
[the remark this answers is not in the document]
Comment (ivanahuckova) [reply public/app/plugins/datasource/prometheus/datasource.ts:252]: Yeah, I agree with @ifrost, I would rephrase to `Exemplars are not available for instant queries.` because there might be a tiny/edge case-y usecase when it makes sense. So I would just say they are not available. Maybe you can check with docs team how would they phrase it.

**Review by ivanahuckova — APPROVED** [review]
LGTM! I left comment regarding phrasing of "we don't support exemplars for instant queries".
